Providers/Profile.swift

```diff
@@ -361,15 +361,15 @@ public class BrowserProfile: Profile {
         }
     }()
 
-    private var dbCreated = false
+    private static var dbCreated = false
```

Comment:
We should add a little block comment here. Something like:

```
/**
 * For safety and efficiency, we only allow each database to be opened once. It
 * looks like you can create multiple Profile instances, each pointing to a
 * different database, but that's not currently the case.
 *
 * These `dbCreated` flags are an implementation detail to avoid opening a DB
 * only to immediately close it -- they reflect whether the singleton DB instances
 * have been assigned.
 */
```

Then we should file a follow-up to make an implementation that either memoizes opens for a particular path (`self.files`), or avoids statics altogether. Things have changed in Swift:

http://stackoverflow.com/questions/24024549/using-a-dispatch-once-singleton-model-in-swift